pkg/scheduler/framework/plugins/clusteraffinity/azure/capacity.go

```diff
@@ -0,0 +1,129 @@
+//
+//Copyright (c) Microsoft Corporation.
+//Licensed under the MIT license.
+
+// Package azure provides Azure-specific utilities and services.
+package azure
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/klog/v2"
+
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	computev1 "go.goms.io/fleet/apis/protos/azure/compute/v1"
+	"go.goms.io/fleet/pkg/clients/azure/compute"
+)
+
+// defaultAzureCapacityService is the default implementation of AzureCapacityService.
+type defaultAzureCapacityService struct {
+	// ComputeClient is the Azure compute client used to validate capacity requirements.
+	ComputeClient compute.AttributeBasedVMSizeRecommenderClient
+}
+
+// Compile-time check to ensure defaultAzureCapacityService implements AzureCapacityService
+var _ AzureCapacityService = &defaultAzureCapacityService{}
+
+// NewAzureCapacityService creates a new default Azure capacity service
+func NewAzureCapacityService(computeClient compute.AttributeBasedVMSizeRecommenderClient) AzureCapacityService {
```

Comment:
I'm sooo sorry for providing the wrong suggestion, but @zhiying-lin taught me this is anti-pattern: https://github.com/Azure/fleet/pull/1213#discussion_r2464514954